[Yip Jung Hon] iP

junghon3709 opened the pull request:
## Tiger Helper Bot

> "Roar" - Tiger [(citation needed)](https://www.google.com/)

Tiger frees your mind so that you have more time to comtemplate how empty your life is. It's
- Buggy
- Hard to learn
- **SLOW** to use

All you need to do is,
- Download another task manager app
- Double click it
- Add your tasks
- Let it manage your tasks for you 😉

And it's **FREE!**

Features:
- [x] There's an app title...
- [x] And absolutely nothing else
- [x] ~~And it works I guess~~ Whoops, guess not! 
- [x] It seems like if you add new checkboxes, it shows up as having more tasks done!
- [x] So here we go...
- [x] Did you know that
- [x] Tigers are the largest amongst other wild cats
- [x] A punch from a Tiger may kill you
- [x] Tigers are nocturnal animals
- [x] Tiger cubs are born blind and only half of the cubs survive
- [x] Tigers love to swim and play in the water
- [x] Tigers live for about 25 years [(Sources)](https://www.ranthamborenationalpark.com/blog/amazing-facts-about-tigers/)



If you are a Java programmer, you can use it to see how **NOT** to write Java. Here's the `main` method:
```java
/**
 * Javadoc so that developers can understand this masterpiece of code.
 *
 * @param args[] I don't know why {@code args[]} is needed.
 */
public static void main(String args[]) {
    boolean isTrue = true;
    if (isTrue == true) {
        System.out.println(isTrue);
    }
}
```

Commit:
Add Gradle support

junghon3709 committed:
Add greeting and bye feature

junghon3709 committed:
Add feature to allow users to add and list tasks

junghon3709 committed:
Add task completion

junghon3709 committed:
Made Task abstract, add ToDofeature, add detection if user double marks task as done

junghon3709 committed:
Refactor code, made each command their own class

junghon3709 committed:
Refactor code to include Command parsing

junghon3709 committed:
Add deadline

junghon3709 committed:
Fix bug, extra spaces added by users before entering command are correctly ignored

junghon3709 committed:
Add events

junghon3709 committed:
Add basic tests

junghon3709 committed:
Add Exceptions DukeEmptyStringException, OutOfBoundsException and InvalidArgumentException

junghon3709 committed:
Add Exception catching to Todo, Event classes

junghon3709 committed:
Add Exception to dateTime classes

junghon3709 committed:
Clean up code, remove second layer of Exception checks which are not needed

junghon3709 committed:
Add support for delete

junghon3709 committed:
Add testing for Level 1 to Leve 6 functionality

junghon3709 committed:
Refactor code

junghon3709 committed:
Rename project from Duke to Tiger

junghon3709 committed:
Update error messages to hide the TigerException class string showing up

junghon3709 committed:
Add process to store taskList

junghon3709 committed:
Add mechanism to load from storage

junghon3709 committed:
Rename classes in parser package, update gitignore to exclude data file

junghon3709 committed:
Refactor code: static Command class handles getting Action from user command

junghon3709 committed:
Make storage functional

junghon3709 committed:
Add datetime class

junghon3709 committed:
Add checks for front back spacing on dates

junghon3709 committed:
Add functionality to parse user input as dates

junghon3709 committed:
Update testing to reflect input dates

junghon3709 committed:
Add testing for dateParser class

junghon3709 committed:
Add testing for DeleteParser

junghon3709 committed:
Add testing for ToDoParser

junghon3709 committed:
Add testing for DeadLineParser

junghon3709 committed:
Add testing for EventParser

junghon3709 committed:
Add testing for MarkDone

junghon3709 committed:
Add some ToDos

junghon3709 committed:
Add javadocs

junghon3709 committed:
Delete unnecessary javadoc files

junghon3709 committed:
Refactor code

junghon3709 committed:
Add find command

junghon3709 committed:
Replace error messages with enums

junghon3709 committed:
Ban users from entering semicolons

junghon3709 committed:
Improve test to test find command

junghon3709 committed:
Add future Todos

junghon3709 committed:
Refactor code

junghon3709 committed:
Add pipeline

junghon3709 committed:
Add partial loading

junghon3709 committed:
Add support for similar commands

junghon3709 committed:
Add todo priority

junghon3709 committed:
Add event priority

junghon3709 committed:
Add deadline priority

junghon3709 committed:
Update test to reflect new list output

junghon3709 committed:
Update test to reflect priority

junghon3709 committed:
Merge branch 'master' into add-gradle-support

junghon3709 committed:
Add gradle support

junghon3709 committed:
Add unit test

junghon3709 committed:
Add javadoc, fix code style

junghon3709 committed:
Minor test fixes

junghon3709 committed:
Add GUI

junghon3709 committed:
Clean up unnecessary parameter in AppState, make app exit gracefully

junghon3709 committed:
Fix bug where users can still add semicolons despite being banned

junghon3709 committed:
Fix bug regarding partial loading

junghon3709 committed:
Change avatars for user and bot

junghon3709 committed:
Add Victor Mono as default font

rickyaandrew commented on src/main/java/Launcher.java:
```diff
@@ -0,0 +1,7 @@
+import javafx.application.Application;
+
+public class Launcher {
+    public static void main(String[] args) {
+        Application.launch(Main.class, args);
+    }
+}
```
Cool code overall sick and neat

rickyaandrew reviewed (commented):
Code is very well done. Good job. Better than mine

rickyaandrew commented on src/main/java/Tiger.java:
```diff
@@ -0,0 +1,32 @@
+        return ui.isExited();
+    }
+
+}
```
i like how your main tiger.java file is this neat. don't miss out javadocs as well

rickyaandrew commented on src/main/java/tiger/actions/Action.java:
```diff
@@ -0,0 +1,11 @@
+package tiger.actions;
+
+import tiger.app.AppState;
+
+/**
+ * The abstract {@code Action} class is responsible for executing the command the user specifies.
+ */
+
+public abstract class Action {
+    public abstract AppState run();
+}
```
inheritance is neat and sick

rickyaandrew commented on src/main/java/tiger/command/Command.java:
```diff
@@ -0,0 +1,118 @@
+        case "done":
+            MarkDoneParser markDoneParser = new MarkDoneParser(command);
+            markDoneParser.parse();
+            return new MarkDoneAction(applicationState, markDoneParser.getIndex() - 1);
+        case "remove":
+            // Fallthrough
+        case "del":
+            // Fallthrough
+        case "delete":
+            DeleteParser deleteCommand = new DeleteParser(command);
+            deleteCommand.parse();
+            return new DeleteAction(applicationState, deleteCommand.getIndex() - 1);
+        case "todo":
+            ToDoParser toDoCommand = new ToDoParser(command);
+            toDoCommand.parse();
+            return new ToDoAction(applicationState, toDoCommand.getTodo(), toDoCommand.getPriority());
+        case "dateline":
+            // Fallthrough;
+        case "deadline":
+            DeadLineParser deadLineCommand = new DeadLineParser(command);
+            deadLineCommand.parse();
+            return new DeadLineAction(applicationState, deadLineCommand.getTodo(), deadLineCommand.getDate(),
+                    deadLineCommand.getPriority());
+        case "event":
+            EventParser eventCommand = new EventParser(command);
+            eventCommand.parse();
+            return new EventAction(applicationState, eventCommand.getTodo(), eventCommand.getDate(),
+                    eventCommand.getPriority());
+        case "search":
+            // Fallthrough
+        case "find":
+            FindParser findCommand = new FindParser(command);
+            findCommand.parse();
+            return new FindAction(applicationState, findCommand.getToSearchFor());
+        case "clear":
+            return new ClearAction(applicationState);
+        default:
+            return new InvalidAction(applicationState);
+        }
+    }
```
like how you have considered so many scenarios

junghon3709 replied:
Thank you for the kind words!

rickyaandrew reviewed (commented):
giving another review as did not went through the entire code the first run. Overall is a very clean code, and have considered many different cases which are not mentioned in the iP

junghon3709 committed:
Add the most sarcastic user guide ever

junghon3709 committed:
Add links to user guide

junghon3709 committed:
Add lines to improve format of docs

junghon3709 committed:
Change docs

junghon3709 committed:
Rewrite docs

junghon3709 committed:
Remove useless imports

junghon3709 committed:
Merge pull request #1 from junghon3709/branch-A-Assertions

Remove useless imports

junghon3709 committed:
Refactor code to allow for exception to take in no arguments

Previously, several exceptions took in an empty string even though the output message did not depend on the string.

To remedy this, the constructors of several exception functions were changed so that it accepts no arguments, if the arguments are not used in the message the exception generations.

This improves the readability of the code.

junghon3709 committed:
Rename variables so to make code clearer

junghon3709 committed:
Merge pull request #2 from junghon3709/branch-A-CodeQuality

Improve code quality

junghon3709 committed:
Add streams

junghon3709 committed:
Merge pull request #3 from junghon3709/branch-A-Streams

Add streams

junghon3709 committed:
Add new gradle builds

Gradle builds previously would only build main class of Launcher.

One had no way to build the main class of Tiger, which is needed to do ui-testing.

Adding builds to gradle to specify which main classes to build will allow us to test the app more easily.

junghon3709 committed:
Add grade build

junghon3709 committed:
Merge pull request #4 from junghon3709/branch-A-Streams

Add gradle support for multiple main classes

junghon3709 committed:
Add CI

Currently, there is no CI being done.

This needs to change so that everytime one does a PR or git push, the new changes do not break the code.

As such, a Github workflow is added so that it builds, runs the unit tests, and runs the text ui tests every time a Github action is take.

This is to minimise bugs.

junghon3709 committed:
Remove checkstyle for github workflow

junghon3709 committed:
Merge pull request #5 from junghon3709/branch-A-CI

Set up continuous integration

junghon3709 committed:
Fix bug in build.gradle, add ability to search by task priority

junghon3709 committed:
Merge pull request #6 from junghon3709/branch-C-Priority

Fix bug in build.gradle, add ability to search by task priority

junghon3709 committed:
Add comments to github workflows and build.gradle

junghon3709 committed:
Merge pull request #7 from junghon3709/branch-C-Priority

Add comments to github workflows and build.gradle

junghon3709 committed:
Fix spacing for printing out taskList

junghon3709 committed:
Fix bug when no tasks are found

When the user enters in a search term, and no tasks are returned, Tiger previously responsed with "You currently have no tasks".

This is wrong and confusing as it implies the user has no tasks, rather than that no tasks match the find description.

This new commit fixes that by first checking if the taskList returned when the user searches for tasks is empty. If it is empty, Tiger responds instead with "I couldn't find any tasks."

junghon3709 committed:
Add text-ui-testing for new priority command

junghon3709 committed:
Update Readme to include searching by priority

junghon3709 committed:
Update image to Ui.png

junghon3709 committed:
Update Readme

junghon3709 committed:
Update Readme

junghon3709 committed:
Set theme jekyll-theme-midnight

junghon3709 committed:
Set theme jekyll-theme-architect

junghon3709 committed:
Set theme jekyll-theme-midnight

junghon3709 committed:
Update Readme

junghon3709 committed:
Merge branch 'master' of https://github.com/junghon3709/ip

junghon3709 committed:
Push full stops inside code to make it render better with midnight theme

junghon3709 committed:
Update readme

junghon3709 committed:
Reformat code according to checkstyle

junghon3709 committed:
Add installation/uninstallation instructions

junghon3709 committed:
Update Ui.png to be in docs folder

junghon3709 committed:
Fix typo in docs
